sdk/cosmosdb/cosmos/test/internal/unit/partitionMerge.spec.ts

```diff
@@ -0,0 +1,218 @@
+
+  const mockDocument1 = createMockDocument(
+    "sample-id-1",
+    "Sample Document 1",
+    "This is the first sample document",
+  );
+  const mockDocument2 = createMockDocument(
+    "sample-id-2",
+    "Sample Document 2",
+    "This is the second sample document",
+  );
+
+  // Define a stub for queryFeed in clientContext
+  sinon.stub(clientContext, "queryFeed").resolves({
+    result: [mockDocument1, mockDocument2] as unknown as Resource, // Add result to mimic expected structure
+    headers: {
+      "x-ms-request-charge": "3.5", // Example RU charge
+      "x-ms-continuation": "token-for-next-page", // Continuation token for pagination
+    },
+    code: 200, // Optional status code
+  });
+
+  // Create a new instance of TestParallelQueryExecutionContext
+  const context = new TestParallelQueryExecutionContext(
+    clientContext,
+    collectionLink,
+    query,
+    options,
+    partitionedQueryExecutionInfo,
+    correlatedActivityId,
+  );
+  context["options"] = options;
+
+  it("there should be 2 document producers in the priority queue as there are two partition key ranges", async function () {
+    // Assert that the priority queue has 2 document producers
+    assert.equal(context["orderByPQ"].size(), 2);
+
+    // Assert that the document producers have the correct start and end EPKs and populateEpkRangeHeaders is false
+    context["orderByPQ"].forEach((docProd) => {
+      if (docProd.targetPartitionKeyRange.id === mockPartitionKeyRange1.id) {
+        assert.equal(docProd.startEpk, mockPartitionKeyRange1.minInclusive);
+        assert.equal(docProd.endEpk, mockPartitionKeyRange1.maxExclusive);
+      } else if (docProd.targetPartitionKeyRange.id === mockPartitionKeyRange2.id) {
+        assert.equal(docProd.startEpk, mockPartitionKeyRange2.minInclusive);
+        assert.equal(docProd.endEpk, mockPartitionKeyRange2.maxExclusive);
+      }
+      assert.equal(docProd.populateEpkRangeHeaders, false);
+    });
+  });
+
+  it("_repairExecutionContext function should be called if partition is gone due to merge", async function () {
+    // Stub the current function of the first document producer to throw a Gone error
+    const docProd1 = context["orderByPQ"].peek();
+    sinon.stub(docProd1, "current").rejects({
+      code: StatusCodes.Gone,
+      substatus: SubStatusCodes.PartitionKeyRangeGone,
+      message: "Partition key range is gone",
+    });
+
+    // Creating a spy on the _repairExecutionContext function
+    const repairSpy = sinon.spy(context as any, "_repairExecutionContext");
+
+    try {
+      // The query fails because the current function of the first document producer throws a Gone error
+      await context.nextItem(context["diagnosticNodeWrapper"]["diagnosticNode"]);
+      assert.fail("Expected query to fail");
+    } catch (err) {
+      assert(err);
+    }
+
+    // Assert that the _repairExecutionContext function was called once
+    assert(repairSpy.calledOnce);
+    repairSpy.restore();
+  });
```

Comment:
Mimic functionality showing end to end case where the parent epk ranges are picked up in the newly created document producers.

Reply:
Modified partitionMerge.spec.ts to handle this case